pkg/proxy/nftables/proxier.go

```diff
@@ -261,7 +265,7 @@ func NewProxier(ipFamily v1.IPFamily,
 }
 
 // Create a knftables.Interface and check if we can use the nftables proxy mode on this host.
-func getNFTablesInterface(ipFamily v1.IPFamily) (knftables.Interface, error) {
+func getNFTablesInterface(ipFamily v1.IPFamily, skipKernelVersionCheck bool) (knftables.Interface, error) {
```

Comment:
in iptables we do this stuff before creating the proxy https://github.com/kubernetes/kubernetes/blob/9791f0d1f39f3f1e0796add7833c1059325d5098/cmd/kube-proxy/app/server_linux.go#L108-L111

should we move this validation to that place instead at the time of creating the proxier?

Reply:
we do it that way for iptables to avoid having identical iptables-constructing functions in `pkg/proxy/iptables` and `pkg/proxy/ipvs`. But really, the logic belongs inside the individual backends; `cmd/kube-proxy` shoudn't be making assumptions about what they need.

Reply:
or alternatively, we could call into a `pkg/proxy/nftables` function from `cmd/kube-proxy` to construct and sanity-check before we actually create the `Proxier`; we sort of do that with ipvs. But that's mostly only advantageous because it lets us avoid separately doing sanity-checking for IPv4 and IPv6, and I have plans to eventually fix that anyway

Reply:
it was  a curious comment 👍